File: frontend/src/modules/visualbim/manifest.ts
import { lazy, type ComponentType, type LazyExoticComponent } from 'react';
import { ScatterChart } from 'lucide-react';
import type { ModuleManifest } from '../_types';

const VisualBimPage = lazy(
  () => import('./VisualBimPage'),
) as unknown as LazyExoticComponent<ComponentType<unknown>>;

export const manifest: ModuleManifest = {
  id: 'visualbim',
  name: 'BIM Analytics',
  description: 'Multi-dimensional point-cloud visualization of BIM project parameters',
  version: '1.0.0',
  icon: ScatterChart,
  category: 'tools',
  defaultEnabled: true,
  routes: [
    {
      path: '/visualbim',
      title: 'BIM Analytics',
      component: VisualBimPage,
    },
  ],
  navItems: [
    {
      labelKey: 'nav.visualbim',
      to: '/visualbim',
      icon: ScatterChart,
      group: 'tools',
    },
  ],
  searchEntries: [
    {
      label: 'BIM Analytics',
      path: '/visualbim',
      keywords: [
        'visualbim', 'analytics', 'bim', 'point cloud', 'scatter', 'plotly',
        'compare', 'snapshot', 'dna', 'parameter cloud',
      ],
    },
  ],
};

File: backend/app/modules/visualbim/manifest.py
"""VisualBIM module manifest."""

from app.core.module_loader import ModuleManifest

manifest = ModuleManifest(
    name="oe_visualbim",
    version="0.1.0",
    display_name="VisualBIM",
    description="Multi-dimensional point-cloud visualization of BIM project parameters.",
    author="NEXUS / O'Neill Contractors",
    category="core",
    depends=["oe_bim_hub"],
    auto_install=True,
    enabled=True,
)
